lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -1226,44 +1353,10 @@ void Inliner::run() {
   for (auto &nla : nlaMap)
     nla.getSecond().applyUpdates();
 
-  llvm::SetVector<Operation *> opsToRemove;
-
-  // Get the driver for this val, and mark the corresponding connect for
-  // removal.
-  auto getDriverAndMarkConnectForRemoval = [&](Value val) -> Value {
-    Value driver;
-    for (Operation *use : val.getUsers())
-      if (auto connect = dyn_cast<FConnectLike>(use))
-        if (connect.getDest() == val) {
-          driver = connect.getSrc();
-          opsToRemove.insert(connect);
-        }
-    return driver;
-  };
-
-  // Ensure that all users of the `wireToRemove` are defined after the driver.
-  // This is required to ensure the driver dominates the users.
-  auto moveUseAfterDef = [&](Operation *wireToRemove, Operation *driver) {
-    for (Operation *user : wireToRemove->getUsers())
-      if (user->isBeforeInBlock(driver))
-        user->moveAfter(driver);
-  };
   // Garbage collect any annotations which are now dead.  Duplicate annotations
   // which are now split.
   for (auto fmodule : circuit.getBodyBlock()->getOps<FModuleOp>()) {
     for (auto &op : *fmodule.getBodyBlock()) {
-      // Remove all the temporary wires of RefType, created during inlining.
-      if (auto wire = dyn_cast<WireOp>(op))
-        if (wire.getResult().getType().isa<RefType>()) {
-          auto driver = getDriverAndMarkConnectForRemoval(wire);
-          if (!driver.isa<BlockArgument>())
-            moveUseAfterDef(wire, driver.getDefiningOp());
-          wire.getResult().replaceAllUsesExcept(
```

Comment:
@prithayan Do you know what the care re:`replaceAllUsesExcept` is for?  Was this an optimization (don't bother updating operations we will remove shortly) or a correctness issue?

Reply:
This was required to handle u-turn, kind of cases, when the module being inlined, drives the output port with an input port. I think you are already handling this case, also there is a lit test for this.

Reply:
Perfect, thanks!